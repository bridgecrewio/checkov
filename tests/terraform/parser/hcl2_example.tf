# Example HCL2 syntax that won't parse in HCL1
# (from https://github.com/virtuald/pyhcl/issues/74)

locals {
  map = {
    r1 = "21.0.0.0/16"
    r2 = "22.4.0.0/16"
  }

  sg = {
    "test" = [
      "r1",
      "r2"
    ]
  }

}

resource "aws_security_group" "test" {
  name   = "test"
  vpc_id = "vpc-xxxxxxxxx"

  dynamic "ingress" {
    for_each = local.sg.test
    content {
      from_port   = 80
      to_port     = 80
      protocol    = "tcp"
      cidr_blocks = split(",", lookup(local.map, ingress.value, ingress.value))
    }
  }
}