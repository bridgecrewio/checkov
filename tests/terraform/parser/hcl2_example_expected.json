{
  "locals": [
    {
      "map": [
        {
          "r1": "21.0.0.0/16",
          "r2": "22.4.0.0/16"
        }
      ],
      "sg": [
        {
          "test": [
            "r1",
            "r2"
          ]
        }
      ]
    }
  ],
  "resource": [
    {
      "aws_security_group": {
        "test": {
          "name": [
            "test"
          ],
          "vpc_id": [
            "vpc-xxxxxxxxx"
          ],
          "dynamic": [
            {
              "ingress": {
                "for_each": [
                  "${local.sg.test}"
                ],
                "content": [
                  {
                    "from_port": [
                      80
                    ],
                    "to_port": [
                      80
                    ],
                    "protocol": [
                      "tcp"
                    ],
                    "cidr_blocks": [
                      "${split(\",\",lookup(local.map,ingress.value,ingress.value))}"
                    ]
                  }
                ]
              }
            }
          ],
          "ingress": [
            "${local.sg.test}"
          ]
        }
      }
    }
  ]
}